src/main/scala/com/microsoft/hyperspace/actions/Action.scala

```diff
@@ -33,6 +33,7 @@ import com.microsoft.hyperspace.telemetry.{AppInfo, HyperspaceEvent, HyperspaceE
  */
 trait Action extends HyperspaceEventLogging with Logging with ActiveSparkSession {
   protected val baseId: Int = logManager.getLatestId().getOrElse(-1)
+  protected def endId: Int = baseId + 2
```

Comment:
Sorry, why +2, can you please explain ?

Reply:
it's because we write the log twice at begin() using `baseId + 1` and end()  using `baseId + 2`. 
As begin() has in-progress state (e.g. creating), Hyperspace always reads the log file written in "end()"

Reply:
nit: Can we rename it to `firstLogEntryId` and `lastLogEntryId`. And add a comment for the same. 
Also should we have an assert that `firstLogEntryId` would always be -1 or an odd number ?

Reply:
baseId is not firstLogEntryId and it requires some change..
If an operation failed in op(), then firstLogEntryId could be an odd number. (though failures are not handled properly now. e.g. #248)

Reply:
cc @apoorvedave1

Reply:
this is required because `LogEntry`.id is not available when creating - `IndexLogEntry`

Reply:
Can we add `final` modifier?